core/src/main/java/brooklyn/catalog/internal/CatalogInitialization.java

```diff
@@ -0,0 +1,380 @@
+                result = catalog.addItems(contents);
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                if (problem==null) problem = e;
+            }
+        }
+        
+        if (result==null && contents!=null && (mode==PopulateMode.XML || mode==PopulateMode.AUTODETECT)) {
+            // then try XML
+            try {
+                populateInitialFromUriXml(catalog, catalogUrl, contents);
+                // clear YAML problem
+                problem = null;
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                if (problem==null) problem = e;
+            }
+        }
+        
+        if (result!=null) {
+            log.debug("Loaded initial catalog from {}: {}", catalogUrl, result);
+        }
+        if (problem!=null) {
+            log.warn("Error importing catalog from " + catalogUrl + ": " + problem, problem);
+            // TODO inform mgmt of error
+        }
+
+    }
+
+    // deprecated XML format
+    @SuppressWarnings("deprecation")
+    private void populateInitialFromUriXml(BasicBrooklynCatalog catalog, String catalogUrl, String contents) {
+        CatalogDto dto = CatalogDto.newDtoFromXmlContents(contents, catalogUrl);
+        if (dto!=null) {
+            catalog.reset(dto);
+        }
+    }
+
+    boolean hasRunAdditions = false;
+    protected void populateAdditions(BasicBrooklynCatalog catalog) {
+        if (Strings.isNonBlank(additionsUri)) {
+            if (disallowLocal) {
+                if (!hasRunAdditions) {
+                    log.warn("CLI additions supplied but not supported in disallow-local mode; ignoring.");
+                }
+                return;
+            }   
+            if (!hasRunAdditions) {
+                log.debug("Adding to catalog from CLI: "+additionsUri+" (force: "+force+")");
+            }
+            Iterable<? extends CatalogItem<?, ?>> items = catalog.addItems(
+                new ResourceUtils(this).getResourceAsString(additionsUri), force);
+            
+            if (!hasRunAdditions)
+                log.debug("Added to catalog from CLI: "+items);
+            else
+                log.debug("Added to catalog from CLI: count "+Iterables.size(items));
+            
+            hasRunAdditions = true;
+        }
+    }
+
+    protected void populateViaCallbacks(BasicBrooklynCatalog catalog) {
+        for (Function<CatalogInitialization, Void> callback: callbacks)
+            callback.apply(this);
+    }
+
+    private Object setFromCLMMutex = new Object();
+    private boolean setFromCatalogLoadMode = false;
+
+    /** @deprecated since introduced in 0.7.0, only for legacy compatibility with 
+     * {@link CatalogLoadMode} {@link BrooklynServerConfig#CATALOG_LOAD_MODE},
+     * allowing control of catalog loading from a brooklyn property */
+    @Deprecated
+    public void applyCatalogLoadMode() {
```

Comment:
probably it could, but just in case the properties are set later i wanted to reflect the original logic. this method should be removed soon (as soon as 0.7.0 is out) so not too concerned.